spec/lib/random_example_spec.rb

```diff
@@ -35,4 +35,31 @@
       }.to raise_error(GovukSchemas::InvalidContentGenerated)
     end
   end
+
+  describe "#customise_and_validate" do
+    it "removes user defined fields for exclusion" do
+      schema = GovukSchemas::Schema.random_schema(schema_type: "notification")
+
+      random_example = GovukSchemas::RandomExample.new(schema: schema).payload
+      # Ensure that the example definitely has the field that the method we are testing will remove.
+      # If we did not add this manually there would only be a 50:50 chance that it would be added by the random generation. This would cause the test to pass by accident 50% of the time.
+      example_with_field = random_example.merge("withdrawn_notice" => {})
+
+      allow_any_instance_of(GovukSchemas::RandomItemGenerator)
+        .to receive(:payload)
+        .and_return(example_with_field)
```

Comment:
That would affect all the fields, no? In which case it might be better to be specific about what we're trying to test.